tests/rptest/tests/rbac_test_v2.py

```diff
@@ -516,3 +519,213 @@ def test_list_user_roles_no_authn(self):
+        ):
+            self.alice_rpk.produce(self.topic0, "foo", "bar")
+
+        self.logger.debug("Now add topic access rights for user")
+        self.su_rpk.sasl_allow_principal(
+            f"RedpandaRole:{self.role_name0}", ["all"], "topic", "*"
+        )
+
+        self.logger.debug(
+            "And a deny ACL to the role which is NOT assigned to the user"
+        )
+        self.su_rpk.sasl_deny_principal(
+            f"RedpandaRole:{self.role_name1}", ["read"], "topic", self.topic1
+        )
+
+        topics = wait_until_result(
+            lambda: self.has_topics(self.alice_rpk),
+            timeout_sec=10,
+            backoff_sec=1,
+            retry_on_exc=True,
+        )
+
+        assert self.topic0 in topics
+        assert self.topic1 in topics
+
+        self.logger.debug("Confirm that the user can produce to both topics")
+
+        self.alice_rpk.produce(self.topic0, "foo", "bar")
+        self.alice_rpk.produce(self.topic1, "baz", "qux")
+
+        self.logger.debug("Confirm that the user can consume both topics")
+
+        rec = json.loads(self.alice_rpk.consume(self.topic0, n=1))
+        assert rec["topic"] == self.topic0, f"Unexpected topic {rec['topic']}"
+        assert rec["key"] == "foo", f"Unexpected key {rec['key']}"
+        assert rec["value"] == "bar", f"Unexpected value {rec['value']}"
+
+        rec = json.loads(self.alice_rpk.consume(self.topic1, n=1))
+        assert rec["topic"] == self.topic1, f"Unexpected topic {rec['topic']}"
+        assert rec["key"] == "baz", f"Unexpected key {rec['key']}"
+        assert rec["value"] == "qux", f"Unexpected value {rec['value']}"
+
+        self.logger.debug(
+            "Now add user to the role with the deny ACL and confirm change in access"
+        )
+
+        _ = self.superuser_admin.add_role_members(role=self.role_name1, members=[alice])
+
+        wait_until(
+            lambda: self.role_for_user(self.role_name1, alice),
+            timeout_sec=10,
+            backoff_sec=1,
+            retry_on_exc=True,
+        )
+
+        wait_until(
+            lambda: "DENY" in self.su_rpk.acl_list(),
+            timeout_sec=10,
+            backoff_sec=1,
+            retry_on_exc=True,
+        )
+
+        with expect_exception(
+            RpkException, lambda e: "AUTHORIZATION_FAILED" in e.stderr
+        ):
+            self.alice_rpk.consume(self.topic1, n=1)
+
+        self.logger.debug(
+            "And finally confirm that the user retains read rights on the other topic"
+        )
+
+        rec = json.loads(self.alice_rpk.consume(self.topic0, n=1))
+        assert rec["topic"] == self.topic0, f"Unexpected topic {rec['topic']}"
+        assert rec["key"] == "foo", f"Unexpected key {rec['key']}"
+        assert rec["value"] == "bar", f"Unexpected value {rec['value']}"
+
+    @cluster(num_nodes=3)
+    @parametrize(delete_acls=True)
+    @parametrize(delete_acls=False)
+    @parametrize(delete_acls=None)
```

Comment:
nit: this turns into 3 different clusters getting spun up and down, consider just having a for loop in the test instead.